tests/Zend/Db/TestSetup.php

```diff
@@ -80,14 +80,18 @@ protected function _setUpTestUtil()
      */
     protected function _setUpAdapter()
     {
-        $this->_db = Zend_Db::factory($this->getDriver(), $this->_util->getParams());
         try {
+            $this->_db = Zend_Db::factory($this->getDriver(), $this->_util->getParams());
             $conn = $this->_db->getConnection();
         } catch (Zend_Exception $e) {
             $this->_db = null;
             $this->assertTrue($e instanceof Zend_Db_Adapter_Exception,
                 'Expecting Zend_Db_Adapter_Exception, got ' . get_class($e));
             $this->markTestSkipped($e->getMessage());
+        } catch (Throwable $e) {
+            if (stristr($e->getMessage(), 'undefined constant')) {
+                $this->markTestSkipped();
+            }
```

Comment:
Example:
- https://github.com/zf1s/zf1/runs/2115567186

But I'm thinking maybe this approach is wrong, better emulate in code the same behavior as older php versions.

Reply:
This class can't be even loaded:
- https://github.com/zf1s/zf1/blob/1cfdc03ff89ddde66d5240c8cd920b725c946ad3/packages/zend-db/library/Zend/Db/Adapter/Db2.php#L82

The exception is thrown on class construction:
- https://github.com/zf1s/zf1/blob/1cfdc03ff89ddde66d5240c8cd920b725c946ad3/packages/zend-db/library/Zend/Db.php#L270

Reply:
I think the only solution here is to assign `$_execute_mode` default value on the constructor. but no idea about is that backward compatible change or not.

Reply:
591a1ab7